assets/app/scripts/services/applicationGenerator.js

```diff
@@ -270,10 +250,12 @@ angular.module("openshiftConsole")
       };
     };
 
-    scope._generateService  = function(input, serviceName, port){
-      if(port === 'None') {
+    scope._generateService  = function(input, serviceName, ports){
+      // Don't generate headless services.
+      if (!ports || !ports.length) {
```

Comment:
This check doesn't mean headless service - only clusterIP: None means headless service.  Change the comment.